File: documind-backend/documind/src/main/java/cn/edu/cdu/documind/entity/Agent.java
package cn.edu.cdu.documind.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 智能体实体类
 * 
 * @author DocuMind Team
 */
@TableName("agent")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Agent {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("user_id")
    private Long userId;

    @TableField("name")
    private String name;

    @TableField("description")
    private String description;

    @TableField("avatar")
    private String avatar;

    // ⭐ 核心字段：用户自定义配置
    @TableField("role_name")
    private String roleName;

    @TableField("system_prompt")
    private String systemPrompt;

    // ⭐ 提示词模式配置
    @TableField("prompt_mode")
    private String promptMode = "single"; // single-单一提示词, multi-多模板, dynamic-动态选择

    // AI 模型配置
    @TableField("model_name")
    private String modelName = "qwen-plus";

    @TableField("temperature")
    private BigDecimal temperature = new BigDecimal("0.70");

    @TableField("max_tokens")
    private Integer maxTokens = 2000;

    @TableField("top_p")
    private BigDecimal topP = new BigDecimal("0.90");

    // RAG 配置
    @TableField("rag_enabled")
    private Integer ragEnabled = 1; // 0-关闭 1-开启

    @TableField("rag_top_k")
    private Integer ragTopK = 10;  // 优化为10，提升检索全面性

    @TableField("rag_similarity_threshold")
    private BigDecimal ragSimilarityThreshold = new BigDecimal("0.50");

    // 统计信息（用于智能体详情页展示）
    @TableField("document_count")
    private Integer documentCount = 0;

    @TableField("chat_count")
    private Integer chatCount = 0;

    @TableField("session_count")
    private Integer sessionCount = 0;

    @TableField("template_count")
    private Integer templateCount = 0;

    @TableField("total_tokens")
    private Long totalTokens = 0L;

    // 时间信息
    @TableField("last_chat_at")
    private LocalDateTime lastChatAt;

    @TableField("status")
    private Integer status = 1; // 0-禁用 1-启用

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;
}

File: documind-frontend/src/api/agent.ts
/**
 * 智能体 API 接口
 *
 * @author DocuMind Team
 */

import { http } from "@/utils/http";

// 智能体类型定义
export interface Agent {
  id: number;
  userId: number;
  name: string;
  description?: string;
  avatar?: string;
  roleName: string;
  systemPrompt: string;
  promptMode: string;
  modelName: string;
  temperature: number;
  maxTokens: number;
  topP: number;
  ragEnabled: number;
  ragTopK: number;
  ragSimilarityThreshold: number;
  documentCount: number;
  chatCount: number;
  sessionCount: number;
  templateCount: number;
  totalTokens: number;
  lastChatAt?: string;
  status: number;
  createdAt: string;
  updatedAt: string;
}

// 创建智能体请求
export interface AgentCreateRequest {
  name: string;
  description?: string;
  avatar?: string;
  roleName: string;
  systemPrompt: string;
  promptMode?: string;
  modelName?: string;
  temperature?: number;
  maxTokens?: number;
  topP?: number;
  ragEnabled?: number;
  ragTopK?: number;
  ragSimilarityThreshold?: number;
}

// 更新智能体请求
export interface AgentUpdateRequest {
  name?: string;
  description?: string;
  avatar?: string;
  roleName?: string;
  systemPrompt?: string;
  promptMode?: string;
  modelName?: string;
  temperature?: number;
  maxTokens?: number;
  topP?: number;
  ragEnabled?: number;
  ragTopK?: number;
  ragSimilarityThreshold?: number;
  status?: number;
}

// 分页响应
export interface PageResponse<T> {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

// 分页查询参数
export interface AgentPageParams {
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortOrder?: string;
  status?: number;
  keyword?: string;
}

/**
 * 创建智能体
 */
export const createAgent = (data: AgentCreateRequest) => {
  return http.request<Agent>("post", "/api/agents", { data });
};

/**
 * 获取智能体列表（不分页）
 */
export const getAgentList = () => {
  return http.request<Agent[]>("get", "/api/agents/list");
};

/**
 * 分页查询智能体
 */
export const getAgentPage = (params: AgentPageParams) => {
  return http.request<PageResponse<Agent>>("get", "/api/agents", { params });
};

/**
 * 获取智能体详情
 */
export const getAgentDetail = (id: number) => {
  return http.request<Agent>("get", `/api/agents/${id}`);
};

/**
 * 更新智能体
 */
export const updateAgent = (id: number, data: AgentUpdateRequest) => {
  return http.request<Agent>("put", `/api/agents/${id}`, { data });
};

/**
 * 删除智能体
 */
export const deleteAgent = (id: number) => {
  return http.request<void>("delete", `/api/agents/${id}`);
};

/**
 * 切换智能体状态
 */
export const toggleAgentStatus = (id: number) => {
  return http.request<Agent>("patch", `/api/agents/${id}/status`);
};

/**
 * 统计智能体数量
 */
export const countAgents = () => {
  return http.request<number>("get", "/api/agents/count");
};
